docs/software_arch.md

```diff
@@ -0,0 +1,75 @@
+The plug-in module is mainly two files:
+
+1. `DeadlineCloudForMaya.mod` -- The [module description file](https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=Maya_SDK_Distributing_Maya_Plug_ins_DistributingUsingModules_ModuleDescriptionFiles_html)
+   for the plug-in.
+2. `plug-ins/DeadlineCloudForMaya.py` -- The implementation of Maya's plug-in interface for the plugin. There is not much to this code.
+   It is just building the shelf in Maya for AWSDeadline, and creating the button(s) in that shelf for the users to click to launch the submitter.
+   Clicking the submitter button calls a function in the `deadline.maya_submitter` package to create the UI, and do all of the submitter functionality.
+
+### `deadline.maya_submitter`
+
+This Python module implements all of the functionality of the Maya plug-in itself. The entrypoints that Maya integrates with
+are all defined in `deadline.maya_submitter.mel_commands`. These simply create a UI window to display to the user.
+
+Fundamentally, what this submitter is doing is creating a [Job Bundle](https://docs.aws.amazon.com/deadline-cloud/latest/developerguide/build-job-bundle.html)
+and using the GUI creation code in the [`deadline` Python package](https://pypi.org/project/deadline/) to generate the UI
+that is displayed to the user. The important parts to know about in a job bundle are:
+
+1. The job template file. The submitter code dynamically generates the template based on properties of the specific scene file
+   that is loaded. For example, it may contain a Step for each layer of the scene to render.
+   Note: All job template files are currently derived from a standard static job template located at
+   `src/deadline/maya_submitter/default_maya_job_template.yaml`.
+2. Asset references. These are the files that the job, when submitted, will require to be able to run. The submitter contains code
+   that introspects the scene that is loaded to automatically discover these, and the submitter UI allows the end-user to modify
+   the list of files.
+
+The job submission itself is handled by functionality within the `deadline` package that is hooked up when the UI is created.
+
+## Maya Adaptor Application
+
+See the [README](../README.md#maya-adaptor) for background on what purpose the adaptor application serves.
+
+The implementation of the adaptor for Maya has two parts:
+
+1. The adaptor application itself whose code is located in `src/deadline/maya_adaptor/MayaAdaptor`. This is the
+   implementation of the command-line application (named `maya-openjd`) that is run by Jobs created by the Maya submitter.
+2. A "MayaClient" application located in `src/deadline/maya_adaptor/MayaClient`. This is an application that is
+   run within Maya itself by the adaptor application when it launches Maya. The MayaClient facilitates communication
+   between the adaptor itself and the running Maya application; communication to tell Maya to, say, load a scene file,
+   or render frame 20 of the loaded scene.
```

Comment:
I think one important detail in here is that the client starts with Maya and runs for the duration of the session, while the adapter runs once per task.

Also a nit: many uses of "itself" which can be removed.

Reply:
reworded.